Sources/PrimerSDK/Classes/Data Models/PaymentMethodTokenizationRequest.swift

```diff
@@ -17,6 +17,7 @@ struct PaymentMethodTokenizationRequest: Encodable {
 
 }
 
+// feels like we could polymorph this with a protocol, or at least restrict construcions with a specific factory method for each payment instrument.
```

Comment:
Yeah. Wanna try sth like the code below?

```
protocol PaymentInstrumentProtocol {
    var name: String { get }
    // ...
}
protocol OAuthPaymentInstrumentProtocol: PaymentInstrumentProtocol {
    var host: String { get }
    var token: String { get }
}

struct CardPaymentInstrument: PaymentInstrumentProtocol {
    var name: String = "Card"
}

struct ApayaPaymentInstrument: OAuthPaymentInstrumentProtocol {
    var name: String = "Apaya"
    var token: String = "access_token"
    var host: String = "https://..."
}
```

Reply:
I'm hesitant to do this because it would entail significant QA overhead. Think it would be better to create a separate PR for this so that the testing is more focused lest we introduce some bug.